src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -148,5 +212,42 @@ private void RemoveStalePools()
             // future, there are a variety of possible ways to address it, such as allowing connections to
             // be returned to pools they weren't associated with.
         }
+
+        internal readonly struct HttpConnectionKey : IEquatable<HttpConnectionKey>
+        {
+            public readonly string Host;
+            public readonly int Port;
+            public readonly string SslHostName;     // null if not SSL
+            public readonly Uri ProxyUri;
+
+            public HttpConnectionKey(string host, int port, string sslHostName, Uri proxyUri)
+            {
+                Host = host;
+                Port = port;
+                SslHostName = sslHostName;
+                ProxyUri = proxyUri;
+            }
+
+            // In the common case, SslHostName (when present) is equal to Host.  If so, don't include in hash.
+            public override int GetHashCode() =>
+                (Host?.GetHashCode() ?? 0) ^
```

Comment:
You can use HashCode.Comine here

Reply:
Thanks, I was trying to remember where this API was...